app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java

```diff
@@ -340,23 +341,44 @@ private void startDecodeQrCodeImages(List<Uri> uris) {
         QrDecodeTask task = new QrDecodeTask(this, (results) -> {
             List<CharSequence> errors = new ArrayList<>();
             List<VaultEntry> entries = new ArrayList<>();
+            List<GoogleAuthInfo.Export> googleAuthExports = new ArrayList<>();
+
             for (QrDecodeTask.Result res : results) {
                 if (res.getException() != null) {
                     errors.add(buildImportError(res.getFileName(), res.getException()));
                     continue;
                 }
 
                 try {
-                    GoogleAuthInfo info = GoogleAuthInfo.parseUri(res.getResult().getText());
-                    VaultEntry entry = new VaultEntry(info);
-                    entries.add(entry);
-                } catch (GoogleAuthInfoException e) {
+                    Uri scanned = Uri.parse(res.getResult().getText());
+                    if (Objects.equals(GoogleAuthInfo.getUriScheme(scanned), GoogleAuthInfo.SCHEME_EXPORT)) {
+                        GoogleAuthInfo.Export export = GoogleAuthInfo.parseExportUri(scanned);
+                        for (GoogleAuthInfo info: export.getEntries()) {
+                            VaultEntry entry = new VaultEntry(info);
+                            entries.add(entry);
+                        }
+                        googleAuthExports.add(export);
+                    } else {
+                        GoogleAuthInfo info = GoogleAuthInfo.parseUri(res.getResult().getText());
+                        VaultEntry entry = new VaultEntry(info);
+                        entries.add(entry);
+                    }
+                } catch (GoogleAuthInfoException | IllegalArgumentException e) {
                     errors.add(buildImportError(res.getFileName(), e));
                 }
             }
 
             final DialogInterface.OnClickListener dialogDismissHandler = (dialog, which) -> importScannedEntries(entries);
-            if ((errors.size() > 0 && results.size() > 1) || errors.size() > 1) {
+            if (!googleAuthExports.isEmpty()) {
```

Comment:
What should happen if I select a couple of images for Google Authenticator export batch, and a couple of regular QR codes? Currently, the latter is not imported.

Reply:
With the revised approach every valid token found in any of the images is in the entries list, if there is a partial export found then details of any scanning errors can be shown as additional details in the dialog and the user can optionally import both the partial export and the regular QR entries. If no dialogs relating to GoogleAuth import are needed then it falls back to the existing Dialog hierarchy.

Another approach could be separating them into 2 separate lists and showing 2 dialogs but I'm not sure if that would be more or less confusing.